### PR TITLE
included oauth security

### PR DESCRIPTION
**Pull request by nilavghosh**
- Included OAuth bearer token security for the predict function
- Reference : https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/
- Added the token endpoint
- No DB added as of now
- changed config paths to absolute paths
- added requirements.txt

### CODE REVIEW

File: autodeploy/deploy.py [hunk 1]
```diff
@@ -144,11 +145,12 @@ def run(self):
 
 
 def main():
+  print(args.config)
```
Comment (kartik4949) [inline autodeploy/deploy.py:148]: lets use logger here

File: autodeploy/routers/security.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from security.model import User, UserInDB
+
+router = APIRouter(
+    prefix="/token",
+    tags=["security"],
+    responses={}
+)
+
+@router.post("/")
+async def login(form_data: OAuth2PasswordRequestForm = Depends()):
```
Comment (kartik4949) [inline autodeploy/routers/security.py:19]: it depends on nothing?

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+requests
+uvicorn
+fastapi
+scikit-learn
```
Comment (kartik4949) [inline requirements.txt:4]: versions??

**Comment by kartik4949** [comment]
also @nilavghosh can we put model.py and schema in autodeploy/model and autodeploy/schema

File: autodeploy/routers/security.py (hunk 2 above)
Comment (nilavghosh) [reply autodeploy/routers/security.py:19]: Yeah! it doesnt. more info here : https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/

File: requirements.txt (hunk 3 above)
Comment (nilavghosh) [reply requirements.txt:4]: been lazy.. let me puth them in

File: autodeploy/routers/predict.py [hunk 4]
```diff
@@ -44,7 +44,7 @@ def setupRabbitMq(self, ):
 
     # connect to RabbitMQ Server.
     connection = pika.BlockingConnection(
-        pika.ConnectionParameters('localhost'))
+        pika.ConnectionParameters('127.0.0.1', port=5672))
```
Comment (kartik4949) [inline autodeploy/routers/predict.py:47]: i forgot to make it configurable

Comment (nilavghosh) [reply autodeploy/routers/predict.py:47]: yeah .. could be in the config.yaml?

File: autodeploy/deploy.py (hunk 1 above)
Comment (nilavghosh) [reply autodeploy/deploy.py]: Oops.. not required.. removed it.

File: autodeploy/routers/security.py (hunk 2 above)
Comment (kartik4949) [reply autodeploy/routers/security.py:19]: cool :)

File: autodeploy/routers/security.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+
+from pydantic import BaseModel
```
[the remark this answers is not in the document]
Comment (nilavghosh) [reply autodeploy/routers/security.py]: Yes. fixed now.

File: autodeploy/routers/predict.py (hunk 4 above)
Comment (kartik4949) [reply autodeploy/routers/predict.py:47]: yeah thats what i was thinking, also Ill make a base class and there this will be default

**Comment by kartik4949** [comment]
@nilavghosh also we can delete .npy file too

**Pull request merged**